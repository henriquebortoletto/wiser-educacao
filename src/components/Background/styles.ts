import styled from "styled-components";

export const Background = styled.section`
  background-image: url(/img/imagem.jpg);
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  width: 100%;
  height: 100%;
  position: relative;
  grid-column: 1/2;

  @media (max-width: 650px) {
    grid-column: auto;
    position: absolute;
    z-index: -1;
  }
`;
export const BackgroundOverlay = styled.div`
  background-image: linear-gradient(to top, #130525, rgba(105, 57, 153, 0));
  display: block;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 100;
`;
