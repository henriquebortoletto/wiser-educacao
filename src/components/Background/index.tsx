import * as S from "./styles";

export default function Background() {
  return (
    <S.Background>
      <S.BackgroundOverlay />
    </S.Background>
  );
}
