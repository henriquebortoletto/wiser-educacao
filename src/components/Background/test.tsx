import { render } from "@testing-library/react";

import Background from ".";

describe("<Background />", () => {
	it("should render the background image correct", () => {
		const { container } = render(<Background />);
		expect(container.firstChild).toHaveStyle(
			`background-image: url(/img/imagem.jpg)`
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
