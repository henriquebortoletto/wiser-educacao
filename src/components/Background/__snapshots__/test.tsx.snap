// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background /> should render the background image correct 1`] = `
<section
  class="styles__Background-sc-1qgc544-0 jnnmli"
>
  <div
    class="styles__BackgroundOverlay-sc-1qgc544-1 gvMmtm"
  />
</section>
`;
